### ISSUE TITLE
CrossBundlesResourceType causes fatal error in overridden ResourceTypeRepository

### ISSUE THREAD

**Issue by mglaman** [jsdrupal/drupal-admin-ui#205]
Inside of ResourceTypeRepository, there is the following code:

```
        $resource_type = new CrossBundlesResourceType(
          $entity_type_id,
          $entity_type_id,
          $entity_type->getClass(),
          static::shouldBeInternalResourceType($entity_type)
        );
```

The problem here is that the ResourceType requires a bundle, and the code is passing in the entity type ID. This is _invalid_ for entities which have a bundle.

If any entity implementing `::bundleFieldDefinitions` loads its bundle, a fatal error occurs. See this in Commerce:

```
  /**
   * {@inheritdoc}
   */
  public static function bundleFieldDefinitions(EntityTypeInterface $entity_type, $bundle, array $base_field_definitions) {
    /** @var \Drupal\commerce_order\Entity\OrderItemTypeInterface $order_item_type */
    $order_item_type = OrderItemType::load($bundle);
    $purchasable_entity_type = $order_item_type->getPurchasableEntityTypeId();
```

The order item typ is null. Leading to:

```
PHP Fatal error:  Uncaught Error: Call to a member function getPurchasableEntityTypeId() on null in /var/www/html/web/modules/contrib/commerce/modules/order/src/Entity/OrderItem.php:440
```

**Comment by mglaman** [comment]
My workaround was to attempt: [Support ResourceType which may have a null bundle defined.](https://www.drupal.org/project/jsonapi/issues/2960766#comment-12685668).

Wim just replied, and I have not gotten back to address.

**Comment by lauriii** [comment]
Solved by #213
